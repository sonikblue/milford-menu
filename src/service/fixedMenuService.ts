import type { Menu } from "@/types/menu";
import type { MenuService } from "./menuService";

const fixedMenu = {
  mains: [
    {
      description: "Pasta bolognese",
    },
    {
      description: "Jacket potato with cheese, beans or tuna",
    },
    {
      description: "Cheese & ham panini",
    },
    {
      description: "Veggie pasta bolognese",
    },
  ],
  desserts: [
    {
      description: "Cinnamon roll",
    },
  ],
};

export class FixedMenuService implements MenuService {
  public todaysMenu(): Menu {
    return fixedMenu;
  }

  public menuFor(): Menu {
    return fixedMenu;
  }
}
