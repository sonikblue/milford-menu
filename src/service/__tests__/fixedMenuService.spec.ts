import { describe, expect, it } from "vitest";
import { FixedMenuService } from "../fixedMenuService";
import type { MenuService } from "../menuService";

describe("FixedMenuService", () => {
  it("Should return a fixed value for today's menu", () => {
    const fixedMenuService: MenuService = new FixedMenuService();

    const menu = fixedMenuService.todaysMenu();

    expect(menu).toMatchSnapshot();
  });

  it("Should return a fixed value for any menu date", () => {
    const fixedMenuService: MenuService = new FixedMenuService();

    const menu = fixedMenuService.menuFor(new Date("2020-03-07"));

    expect(menu).toMatchSnapshot();
  });
});
