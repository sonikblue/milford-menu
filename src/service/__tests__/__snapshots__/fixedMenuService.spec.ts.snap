// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FixedMenuService > Should return a fixed value for any menu date 1`] = `
{
  "desserts": [
    {
      "description": "Cinnamon roll",
    },
  ],
  "mains": [
    {
      "description": "Pasta bolognese",
    },
    {
      "description": "Jacket potato with cheese, beans or tuna",
    },
    {
      "description": "Cheese & ham panini",
    },
    {
      "description": "Veggie pasta bolognese",
    },
  ],
}
`;

exports[`FixedMenuService > Should return a fixed value for today's menu 1`] = `
{
  "desserts": [
    {
      "description": "Cinnamon roll",
    },
  ],
  "mains": [
    {
      "description": "Pasta bolognese",
    },
    {
      "description": "Jacket potato with cheese, beans or tuna",
    },
    {
      "description": "Cheese & ham panini",
    },
    {
      "description": "Veggie pasta bolognese",
    },
  ],
}
`;
